src/main/java/frc/robot/subsystems/DriveTrainSubsystem.java

```diff
@@ -65,49 +70,60 @@ public DriveTrainSubsystem() {
     m_pidControllerRight = m_rightLead.getPIDController();}
 
 
-  private AHRS m_navX; 
+  private double krightMinOutput; 
   /** Creates a new DriveTrainSubsystem. */
   public DriveTrainSubsystem(AHRS navX) {
     //Motor controlers
     m_leftLead.restoreFactoryDefaults();
     m_leftFollow.restoreFactoryDefaults();
+    m_leftEncoder = m_leftLead.getEncoder();
+
     m_rightLead.restoreFactoryDefaults();
     m_rightFollow.restoreFactoryDefaults();
+    m_rightEncoder = m_rightLead.getEncoder();
+
     m_leftFollow.follow(m_leftLead);
     m_rightFollow.follow(m_rightLead);
+
     m_pidControllerLeft = m_leftLead.getPIDController();
     m_pidControllerRight = m_rightLead.getPIDController();
 
     m_navX = navX;
+
+    // Setting up the odometry object in the constructor--A little sketchy? No errors and it builds though
+    m_odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(robotPitch()), Units.feetToMeters(getRightEncoderFeet()), Units.feetToMeters(getLeftEncoderFeet()));
```

Comment:
Nothing wrong with setting up member variables in a constructor. It's pretty common particularly if you need something else set up first (like the NavX in this case).

Reply:
It is usually a bad idea to set up member variables in methods that are not the constructor though. That leads to half-initialized classes that can break in strange ways if whatever method sets up the variable isn't guaranteed to be called before a method that uses it.